terraform/modules/container-definition/variables.tf

```diff
@@ -19,7 +19,7 @@ DESC
 }
 
 variable "dependsOn" {
-  type        = list(any)
+  type        = list(object({ containerName = string, condition = string }))
```

Comment:
In this case though, there's only a single list element (type) of `object` - is the problem that this object contains `string` properties?